docs: add intro and edit ignoring files page (#16510)
eslint/eslint: docs/src/user-guide/configuring/ignoring-code.md

---

> You can configure ESLint to ignore certain files and file patterns when linting.

Reviewer: I'm not sure if it is correct to say "to ignore file patterns". Files are ignored, file patterns are a way to specify which files are ignored.

Reviewer 2: Agree, maybe something like this?

```suggestion
You can configure ESLint to ignore certain files and directories while linting by specifying one or more glob patterns.
```

Reviewer 3: This looks better 👍🏻

eslint --ignore-path .gitignore file.js
```

> Any file that follows the standard ignore file format can be used. Keep in mind that specifying `--ignore-path` means that any existing `.eslintignore` file are not used. Note that globbing rules in `.eslintignore` follow those of `.gitignore`.

Reviewer 4: ```suggestion
Any file that follows the standard ignore file format can be used. Keep in mind that specifying `--ignore-path` means that the existing `.eslintignore` file is not used. Note that globbing rules in `.eslintignore` follow those of `.gitignore`.
```

Only one `.eslintignore` file can be used at a time right?

## Using eslintIgnore in package.json

> If an `.eslintignore` file is not found and an alternate file is not specified, ESLint looks in package.json for an `eslintIgnore` key to check for files to ignore.

Author: ```suggestion
If an `.eslintignore` file is not found and an alternate file is not specified, ESLint looks in `package.json` for the `eslintIgnore` key to check for files to ignore.
```

## Ignored File Warnings

> When you pass directories to ESLint, files and directories are silently ignored. If you pass a specific file to ESLint, then ESLint creates a warning that the file was skipped. For example, suppose you have an `.eslintignore` file that looks like this:

Author: 'an' is correct b/c '.eslintrc' starts w vowel sound.

> When you pass directories to ESLint, files and directories are silently ignored. If you pass a specific file to ESLint, then ESLint creates a warning that the file was skipped. For example, suppose you have an `.eslintignore` file that looks like this:

Reviewer 3: Depends on how we read it. I'm reading it as `Dot ESLint rc` not sure if that's correct.

> When you pass directories to ESLint, files and directories are silently ignored. If you pass a specific file to ESLint, then ESLint creates a warning that the file was skipped. For example, suppose you have an `.eslintignore` file that looks like this:

Reviewer 2: I usually don’t say “dot”. I don’t think this matters too much so long as it’s consistent for both eslintrc and eslintignore.

> When you pass directories to ESLint, files and directories are silently ignored. If you pass a specific file to ESLint, then ESLint creates a warning that the file was skipped. For example, suppose you have an `.eslintignore` file that looks like this:

Reviewer 3: In the same file above we have used "a", under the `## The .eslintignore File` section [here](https://github.com/eslint/eslint/pull/16510/files#diff-a336b2f4fd9fc95d4c173e4d02d9832abfbbec335c0ca3cf935865d9fd577e22R47) and [here too](https://github.com/eslint/eslint/pull/16510/files#diff-a336b2f4fd9fc95d4c173e4d02d9832abfbbec335c0ca3cf935865d9fd577e22R41).

Let's go with "a" for consistency.

> When you pass directories to ESLint, files and directories are silently ignored. If you pass a specific file to ESLint, then ESLint creates a warning that the file was skipped. For example, suppose you have an `.eslintignore` file that looks like this:

Author: based on a quick search across the docs source files, we are actually evenly split with how we use (9 of each). 

will make a separate PR to be consistent w use, and will make that "an"
